### PR TITLE
docs(themes): Update working-with-themes.md

### LINKED ISSUE #11544: Documentation of theming support
## What would you like clarification on:

How to use light/dark theme in Uno Platform:

- How is the default theme determined (OS theme support)
- How to switch OS theme for each platform (e.g. action bar on Android, or Settings on iOS)
- How to force a specific theme Application-wide
- How to switch theme at runtim
- Current limitations (element-level theming)
- **See comments below for further points**

## Concern?

- [ ] Usage in industry
- [ ] Clarification of capabilities
- [ ] Getting started with Uno
- [x] Developing with Uno
- [ ] Contributing to the Uno project
- [ ] Publishing your application
- [ ] Support
- [ ] Other (please specify):

## For which Platform:

All

### PR DESCRIPTION
**Pull request by lukeblevins**
GitHub Issue (If applicable): closes #11544 

## PR Type

What kind of change does this PR introduce?
- Documentation content changes

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Discussed with @nickrandolph and @MartinZikmund 

Internal Issue (If applicable):

### CODE REVIEW

File: doc/articles/features/working-with-themes.md [hunk 1]
```diff
@@ -2,30 +2,114 @@
...
 
-As in UWP, the default theme is `Dark`, but in the case of Uno, we use `Light` for all supported systems, as that is the default on all except Windows. However, if you don't specify a theme in the constructor of `App` and don't specify it in `AppResources.xaml` or `App.xaml` either, it will be determined by the current system theme on iOS, Android and to current system/browser theme (depending on browser implementation) on WASM.
+These themes affect both the `Background` and `Foreground` colors to accomodate user preferences. All of the color modes mentioned above are available for use in your app. This guide will detail how to change the system theme setting, make your app recieve change notifications for it, and react to those changes at runtime.
```
Comment (agneszitte) [inline doc/articles/features/working-with-themes.md:9]:
```suggestion
These themes affect both the `Background` and `Foreground` colors to accommodate user preferences. All of the color modes mentioned above are available for use in your app. This guide will detail how to change the system theme setting, make your app receive change notifications for it, and react to those changes at runtime.
```

File: doc/articles/features/working-with-themes.md [hunk 2]
```diff
@@ -2,30 +2,114 @@
...
 
-To set `HighContrast` theme or a custom theme, you can use the `Uno.UI.ApplicationHelper` class:
+## Enable dark mode
 
+As in WinUI, the possible values `Light`, `Dark`, and `HighContrast` correspond to a value users can select in the settings of their respective platform.
```
Comment (agneszitte) [inline doc/articles/features/working-with-themes.md:13]:
```suggestion
As in WinUI, the possible values `Light`, `Dark`, and `HighContrast` correspond to a value users can select in the settings of their respective platforms. 
```

File: doc/articles/features/working-with-themes.md [hunk 3]
```diff
@@ -2,30 +2,114 @@
...
+These themes affect both the `Background` and `Foreground` colors to accomodate user preferences. All of the color modes mentioned above are available for use in your app. This guide will detail how to change the system theme setting, make your app recieve change notifications for it, and react to those changes at runtime.
 
-To set `HighContrast` theme or a custom theme, you can use the `Uno.UI.ApplicationHelper` class:
+## Enable dark mode
 
+As in WinUI, the possible values `Light`, `Dark`, and `HighContrast` correspond to a value users can select in the settings of their respective platform. 
+
+High Contrast mode is often available to enable seperately, from a dedicated _Accessibility_ page. For the purpose of this documentation, we will assume the user wants to use Dark mode.
```
Comment (agneszitte) [inline doc/articles/features/working-with-themes.md:15]:
```suggestion
High Contrast mode is often available to enable separately, from a dedicated _Accessibility_ page. For the purpose of this documentation, we will assume the user wants to use Dark mode.
```

File: doc/articles/features/working-with-themes.md [hunk 4]
```diff
@@ -2,30 +2,114 @@
...
+    }
+}
 ```
-Uno.UI.ApplicationHelper.RequestedCustomTheme = "HighContrast";
-```
 
-This must be called during app startup in the `App` class constructor. Uno does not yet support `FrameworkElement.RequestedTheme` so the theme cannot be changed dynamically at runtime.
+## Change the app theme at runtime
+
+The `SetApplicationTheme` method in this helper class can also be used to support an in-app toggle for dark mode. For example, you could add a toggle button to your app's settings page that allows the user to switch between light and dark mode. The following code snippet shows how to implement this:
```
Comment (agneszitte) [inline doc/articles/features/working-with-themes.md:74]:
```suggestion
The `SetApplicationTheme` method in this helper class can also be used to support an in-app toggle for dark mode. For example, you could add a toggle button to your app's settings page that allows the user to switch between light and dark modes. The following code snippet shows how to implement this:
```

File: doc/articles/features/working-with-themes.md [hunk 5]
```diff
@@ -2,30 +2,114 @@
...
+    private void OnDarkModeToggleToggled(object sender, RoutedEventArgs e)
+    {
+        SystemThemeHelper.SetApplicationTheme(darkMode: this.DarkModeToggle.IsOn);
+    }
+}
 ```
+
+## Change the app theme at startup
+
+Another method to change the app theme is to adjust it immediately upon startup. The following must only be called in the `App` class constructor
```
Comment (agneszitte) [inline doc/articles/features/working-with-themes.md:105]:
```suggestion
Another method to change the app theme is to adjust it immediately upon startup. The following must only be called in the `App` class constructor.
```

**Review by jeromelaban — APPROVED** [review]

File: doc/articles/features/working-with-themes.md [hunk 6]
```diff
@@ -2,30 +2,114 @@
...
-Uno.UI.ApplicationHelper.RequestedCustomTheme = "HighContrast";
-```
 
-This must be called during app startup in the `App` class constructor. Uno does not yet support `FrameworkElement.RequestedTheme` so the theme cannot be changed dynamically at runtime.
+## Change the app theme at runtime
+
+The `SetApplicationTheme` method in this helper class can also be used to support an in-app toggle for dark mode. For example, you could add a toggle button to your app's settings page that allows the user to switch between light and dark modes. The following code snippet shows how to implement this:
+
+```csharp
+using CommunityToolkit.WinUI.UI.Helpers;
```
Comment (Xiaoy312) [inline doc/articles/features/working-with-themes.md:77]: doesnt seem to be needed for the part of code shown:
```suggestion
using CommunityToolkit.WinUI.UI.Helpers;
```

File: doc/articles/features/working-with-themes.md [hunk 7]
```diff
@@ -2,30 +2,114 @@
...
+    }
+}
 ```
+
+## Change the app theme at startup
+
+Another method to change the app theme is to adjust it immediately upon startup. The following must only be called in the `App` class constructor.
+
+```csharp
+Uno.UI.ApplicationHelper.RequestedCustomTheme = "Dark";
```
Comment (Xiaoy312) [inline doc/articles/features/working-with-themes.md:108]: you should just include the full code (containing method signature, uno vs windows code), so it is clear at a glance:
```suggestion
public App()
{
	this.InitializeComponent();

#if HAS_UNO
	Uno.UI.ApplicationHelper.RequestedCustomTheme = nameof(ApplicationTheme.Dark);
#else
	this.RequestedTheme = ApplicationTheme.Dark;
#endif
}
```

File: doc/articles/features/working-with-themes.md [hunk 8]
```diff
@@ -2,30 +2,114 @@
...
+    }
+}
 ```
-Uno.UI.ApplicationHelper.RequestedCustomTheme = "HighContrast";
-```
 
-This must be called during app startup in the `App` class constructor. Uno does not yet support `FrameworkElement.RequestedTheme` so the theme cannot be changed dynamically at runtime.
+## Change the app theme at runtime
+
+The `SetApplicationTheme` method in this helper class can also be used to support an in-app toggle for dark mode. For example, you could add a toggle button to your app's settings page that allows the user to switch between light and dark modes. The following code snippet shows how to implement this:
```
Comment (Xiaoy312) [inline doc/articles/features/working-with-themes.md:74]: be specific in what you refers to as: "in this helper class"
```suggestion
The [`SystemThemeHelper.SetApplicationTheme`](https://github.com/unoplatform/uno.toolkit.ui/blob/main/src/Uno.Toolkit.UI/Helpers/SystemThemeHelper.cs) method from `Uno.Toolkit.UI` can also be used to support an in-app toggle for dark mode. For example, you could add a toggle button to your app's settings page that allows the user to switch between light and dark modes. The following code snippet shows how to implement this:
```

File: doc/articles/features/working-with-themes.md [hunk 9]
```diff
@@ -2,30 +2,114 @@
...
+## React to OS theme changes
+
+When you change the theme mode on your device, the system will send a notification to your app. The colors of UI elements in your app will automatically switch over as long as you do _not_ specify a theme in any of the following places:
+
+* `App` constructor
+* `App.xaml` 
+* `AppResources.xaml`
+* The `RequestedTheme` property of a parent `FrameworkElement`
+
+However, your app can still react manually to changes in the theme mode. To do so, you can use the [Uno.CommunityToolkit.WinUI.UI](https://www.nuget.org/packages/Uno.CommunityToolkit.WinUI.UI) NuGet package. This package is not required to change the app color theme, but it contains a `ThemeListener` class that can be used to listen for OS theme changes. To actually change the app color theme at runtime, you need to install the [Uno.Toolkit.UI](https://www.nuget.org/packages/Uno.Toolkit.UI) NuGet package which contains a `SystemThemeHelper` class.
```
[the remark this answers is not in the document]
Comment (lukeblevins) [reply doc/articles/features/working-with-themes.md:44]: @kazo0 I agree in principle that requiring a third party library for this rather fundamental feature doesn't have the best look. What should the next steps be, given that the doc now is functionally unhelpful and very out of date? I can open an issue to track exposing the existing code in Uno.UWP from the Toolkit SystemThemeHelper.

Would you consider it a roadblock for this pull request? I feel like getting _something_  out that is improved should be the priority until we expose a proper theme listener in the toolkit.

[the remark this answers is not in the document]
Comment (Xiaoy312) [reply doc/articles/features/working-with-themes.md:44]: We can publish the doc like this for the current state. 
Just add a bullet point to update the doc as part of that issue.

[the remark this answers is not in the document]
Comment (lukeblevins) [reply doc/articles/features/working-with-themes.md:44]: @Xiaoy312 Is this PR ready to approve, then?

**Review by Xiaoy312 — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

**Comment by lukeblevins** [comment]
@jeromelaban Was this PR ready to merge?

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/4.9

**Pull request merged**